src/analysis/retail/hunter/survival/modules/apl/AplCheck.tsx

```diff
@@ -0,0 +1,196 @@
+        hasFocus(10),
+        cnd.or(cnd.buffPresent(SPELLS.DEATHBLOW_BUFF), cnd.inExecute(0.2)),
+      ),
+      (tense) => <>. Do not delay for Tip of the Spear stacks.</>,
+    ),
+  },
+  {
+    spell: TALENTS.WILDFIRE_BOMB_TALENT,
+    condition: cnd.describe(
+      cnd.spellFractionalCharges(TALENTS.WILDFIRE_BOMB_TALENT, { atLeast: 1.9 }),
+
+      (tense) => (
+        <>
+          you {tenseAlt(tense, <>have</>, <>had</>)} {'>'} 1.9 charges
+        </>
+      ),
+    ),
+  },
+  {
+    spell: TALENTS.WILDFIRE_BOMB_TALENT,
+    condition: cnd.describe(
+      cnd.and(
+        cnd.spellAvailable(TALENTS.COORDINATED_ASSAULT_TALENT),
+        cnd.hasTalent(TALENTS.BOMBARDIER_TALENT),
+      ),
+      (tense) => (
+        <>
+          {' '}
+          you {tenseAlt(tense, <>have</>, <>had</>)} Bombardier Talented & Coordinated Assault is
+          Ready
+        </>
+      ),
+    ),
+  },
+  {
+    spell: TALENTS.WILDFIRE_BOMB_TALENT,
+    condition: cnd.describe(
+      cnd.and(
+        cnd.spellFractionalCharges(SPELLS.WILDFIRE_BOMB_IMPACT, { atLeast: 1.5, atMost: 2 }),
+        withTip,
+      ),
+      (tense) => (
+        <>
+          {' '}
+          you {tenseAlt(tense, <>have</>, <>had</>)} at least 1 stack of Tip and {'>'} 1.5 charges
+        </>
+      ),
+    ),
+  },
+  //Explosive goes tipless now thanks to grenade juggler rework in .5
+  TALENTS.EXPLOSIVE_SHOT_TALENT,
+  {
+    spell: TALENTS.RAPTOR_STRIKE_TALENT,
+    condition: cnd.describe(
+      cnd.and(
+        cnd.hasTalent(TALENTS.FURIOUS_ASSAULT_TALENT),
+        cnd.buffPresent(SPELLS.FURIOUS_ASSAULT_BUFF_SV),
+      ),
+      (tense) => <>Furious Assault is up.</>,
+    ),
+  },
+  {
+    spell: TALENTS.KILL_COMMAND_SURVIVAL_TALENT,
+    condition: cnd.describe(
+      cnd.and(
+        cnd.hasResource(RESOURCE_TYPES.FOCUS, { atMost: 77 }),
+        cnd.or(
+          cnd.and(
+            cnd.hasTalent(TALENTS.RELENTLESS_PRIMAL_FEROCITY_TALENT),
+            cnd.buffMissing(SPELLS.COORDINATED_ASSAULT_BUFF),
+          ),
+          cnd.or(
+            cnd.and(
+              cnd.hasTalent(TALENTS.RELENTLESS_PRIMAL_FEROCITY_TALENT),
+              cnd.buffPresent(SPELLS.COORDINATED_ASSAULT_BUFF),
+              cnd.buffMissing(SPELLS.TIP_OF_THE_SPEAR_CAST),
+            ),
+            cnd.hasResource(RESOURCE_TYPES.FOCUS, { atMost: 30 }),
+          ),
+        ),
+      ),
+
+      //kill_command,target_if=min:bloodseeker.remains,if=focus+cast_regen<focus.max&(!buff.relentless_primal_ferocity.up|(buff.relentless_primal_ferocity.up&buff.tip_of_the_spear.stack<1|focus<30))
+      (tense) => (
+        <>
+          you {tenseAlt(tense, <>have</>, <>had</>)} at most 77 focus and don't have
+          <SpellLink spell={TALENTS.RELENTLESS_PRIMAL_FEROCITY_TALENT} /> up OR you have
+          <SpellLink spell={TALENTS.RELENTLESS_PRIMAL_FEROCITY_TALENT} /> with either no Tip stacks
+          or less than 30 focus.
+        </>
+      ),
+    ),
+  },
```

Comment:
this condition might be *too complex* to be useful in practice. it probably warrants splitting up into multiple rules:

- cast kill command to apply coordinated assault (if you won't overcap focus)
- cast kill command to generate tip of the spear stacks (if you won't overcap focus)
- cast kill command to generate focus when you're low

the sequencing of the rules functions like an or, and can be easier to understand than a single large condition. combining a couple can be okay (i think both the tip and CA rules are kinda related? so maybe they go together?), but at 3 clauses its pretty chonky

Reply:
By splitting into multiple rules you mean like I have Wildfire bomb with a 1.9 fractional, 1.5 fractional w/ Tip, and Bombardier is talented & CA is ready right?

Reply:
something like that, yes. the goal is clarity. too many rules can hurt that, so it is a bit of a judgment call. sometimes doing an combined rule is better. my 2c is to focus on *intent* for it. what you're describing with wildfire bomb sounds like:

- don't cap wildfire bomb charges
- spend tip on wildfire bombs before CA

is that right?

Reply:
Don't cap bombs
Dump bombs before CA if you have bombardier talented because bombardier refreshes bomb to 2 charges
Bomb if you're ~1.5 charges and have a tip stack - The reason for this one is if you don't throw it at roughly this point you'll very rapidly approach the 1.9 and cap since we have so much bomb CDR.

Then there is a lower priority wildfire bomb that I haven't implemented yet which accounts for if explosive, killshot, butchery are all on cooldown and you're high focus, to just use the bomb charge since it's still better than raptoring.

The APL itself in simc isn't particularly large but it's largely because they combined a lot of the bomb conditions into single lines since they're all ~ equal.

So if that's the case I think multiple lines and then I'll have to put a section above it discussing that certain priorities are equal despite having an apparent hierarchy in the APL list?

Reply:
like i said, its a judgment call. what do you think is clear and easy to understand? if you were to explain to someone in discord when to push bomb, what would you tell them?

there's room for the checker in wowa to be more forgiving / allow slightly sub-optimal play in pursuit of being easier to understand and still catching a ton of common problems.

you can also do things that combine multiple different spells into one. for example a "dump focus" rule at the bottom of the APL that combines Wildfire Bomb, Raptor Strike, or whatever else as a catch-all "push something, anything is better than nothing" rule

Reply:
(also, it doesn't need to be perfect. i'm prone to trying to make it so, but everything here is iterative. its okay to merge something that has flaws and then adjust later)